zellij-server/src/panes/grid.rs

```diff
@@ -170,6 +177,7 @@ fn transfer_rows_from_viewport_to_lines_above(
             viewport.insert(0, row);
         }
     }
+    transferred_rows_height
```

Comment:
We need to account for the `bounded_push` above somehow. There's a chance that it drops a line if it overflows the buffer... maybe also return a value from it?